src/mlpack/core/data/histogram.hpp

```diff
@@ -0,0 +1,59 @@
+/**
+ * @file core/data/histogram.hpp
+ * @author Aditi Pandey
+ *
+ * Histogram function. The purpose of this function is to plot graph w.r.t 
+ * given x and y list of inputs.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_DATA_HISTOGRAM_HPP
+#define MLPACK_CORE_DATA_HISTOGRAM_HPP
+
+#include <mlpack/prereqs.hpp>
+#include <mlpack/core.hpp>
+
+#include "matplotlibcpp.h"
+namespace plt = matplotlibcpp;
+
```

Comment:
@AditiThirdEye Thanks for your contribution, the idea is good, but there are several flows:
This code adds a direct dependency on an external library `matplotlib` inside mlpack core.
Matplotlibcpp requires the uses to have `matplotlib` python installed too, which will eventually make mlpack library dependent on python.
Our current objective is to remove the dependency on external library, unless it is very necessary with preference header-only.
I am not sure if this is the right place for this feature, nor what is going to add to mlpack library.
We have an example repository `mlpack/examples`, if you would like to show to plot data set using C++, I think `mlpack/examples` is the place to for it, even though the dependency seems to be quite large at the same time.

Reply:
Hello @shrit , 
Actually in this #2722 
In the histogram task it was asked to make use of the matplotlibcpp lib
Hence I used it.

Reply:
Also in the discussion of #2722 
You can find below that @heisenbuug  asked me to use matplotlib.cpp as a single header file
So can you please guide me how can I improve my contribution.

Regards,
Aditi

Reply:
@shrit but if we plan on making something like a data frame we would need
to plot the data while preprocessing to get an insight into the data, like if it's skewed
or to look at outliers.

Should we use something else here? I know the idea is not actually totally clear here
and may require a to of discussion here.

Actually, I was looking into the visualizer project `mlboard` which we had last summer, maybe
we connect to that.

Reply:
I would also just put this into the examples repo, e.g. into the utils folder - https://github.com/mlpack/examples/tree/master/utils. I don't think there is need to add this to the core repo; users can always build mlpack and use utility functions from other sources. In this case if I like to plot the histogram I could use the function from the examples repo, that has some other utility functions that are meant to visualize data.

Reply:
Okay @zoq sure, I'll do that 👍🏼

Reply:
Okay @zoq @shrit.
No need to add these to the core mlpack, that makes sense.

Thank you.